includes/blocks.inc

```diff
@@ -74,23 +74,89 @@ function islandora_basic_collection_clear_object_count_cache() {
  * Outputs a JSON string to the buffer.
  */
 function islandora_basic_collection_object_count_callback() {
+  $count_method = variable_get('islandora_basic_collection_count_method', 'SPARQL');
+
+
   $cid = islandora_basic_collection_get_object_count_block_cache_id();
 
   if ($value = cache_get($cid, 'cache')) {
     $substitutions = $value->data;
   }
   else {
+
     $tuque = islandora_get_tuque_connection();
     $objects_query_array = islandora_basic_collection_get_query_info(array(
       'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
       'collection_listing' => TRUE,
       'all_objects' => TRUE,
     ));
     $collection_query_array = islandora_basic_collection_get_query_info(array(
-      'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
-      'collection_listing' => TRUE,
-      'all_objects' => FALSE,
-    ));
+    'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
+    'collection_listing' => TRUE,
+    'all_objects' => FALSE,
+  ));
+
+
+  // Setup Solr query for collection and object counts.
+  if ($count_method == "Solr" && module_exists('islandora_solr')) {
+    $cmodel_field = variable_get('islandora_solr_content_model_field', 'RELS_EXT_hasModel_uri_ms');
+
+    $all_cmodels = islandora_get_content_models();
+    $models_to_exclude = variable_get('islandora_basic_collection_object_count_listing_content_models_to_restrict', FALSE);
+    $cmodels_to_count = array();
+    foreach ($all_cmodels AS $cmodel) {
```

Comment:
This is not related to your code, but wonder what happens if I actually exclude all cmodels?

Reply:
I just tested that... In the Solr method, it counts everything (don't know why, but cool). In the default SPARQL method, it seems to get stuck on "Counting..."

So the Solr version is a bit of an improvement there, though imperfect. Maybe a later ticket to deal with this.